apps/provisioning_api/lib/Controller/UsersController.php

```
@@ -193,6 +200,18 @@ public function addUser(string $userid, string $password, array $groups = []): D
 			}
 		}
 
+		$generatePasswordResetToken = false;
+		if ($password === '') {
+			if ($email === '') {
+				throw new OCSException('To send a password link to the user an email address is required.', 108);
+			}
+
+			$password = $this->secureRandom->generate(10);
+			// Make sure we pass the password_policy
+			$password .= $this->secureRandom->generate(2, '$!.,;:-~+*[]{}()');
```

well this is only 'kind of' true. We don't know if this will pass the password policy test because we don't know what it is set to.

---

Nothing you can fix here. But still

---

I used old code, so I'm to blame here :D 
To be fair, the goal is to create a secure password that no one will know about since the user will receive a mail to reset it.

---

Then this might fail hard when using the password_policy app ;) But like I said fine by me. Lets do this for now.